src/mixer.c

```diff
@@ -353,36 +354,34 @@ void MIXER_ApplyMixer(struct Mixer *mixer, volatile s32 *raw, s32 *orig_value)
 #if HAS_EXTENDED_AUDIO
     case MUX_BEEP:
         if (orig_value) {
-            s32 new_value = raw[mixer->dest + NUM_INPUTS + 1];
-            if (abs(value - new_value) > 100)
+            if (abs(value - scaled_value) > 100)
                 MIXER_SET_BEEP_LOCK(mixer, 0);
             if (! MIXER_BEEP_LOCK(mixer)) {
-                if ((value > *orig_value && value < new_value) ||
-                    (value < *orig_value && value > new_value) ||
-                    (abs(value - new_value) <= 10)) {
+                if ((value > *orig_value && value < scaled_value) ||
+                    (value < *orig_value && value > scaled_value) ||
+                    (abs(value - scaled_value) <= 10)) {
                     MIXER_SET_BEEP_LOCK(mixer, 1);
-                    MUSIC_Play(MUSIC_MAXLEN);
+                    MUSIC_Play(MUSIC_SAVING);
                 }
```

Comment:
This change of song doesn't seem related to the rest of the patch...why is it here?